lib/heartcheck.rb

```diff
@@ -95,6 +99,13 @@ def info_checks
       checks.select { |ctx| ctx.respond_to?(:info) }
     end
 
+    # an executor class that respond to execute(checkers)
+    #
+    # @return [Heartcheck::Services::ResponseFormatter] or nil
+    def formatter
+      @formatter ||= Heartcheck::Services::ResponseFormatter.new if response_formatter
```

Comment:
let's try not to have this conditional, maybe introduce a default formatter that just returns the input value and override with the hash formatter if needed